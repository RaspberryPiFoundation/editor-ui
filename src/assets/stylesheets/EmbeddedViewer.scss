@use './rpf_design_system/colours' as *;

.embedded-viewer {
  display: flex;
  flex: 1 1 auto;
  block-size: auto;
  overflow: hidden;

  .proj-runner-container {
    display: flex;
    flex: 1 1 auto;
    flex-direction: column;
    border-radius: 10px;
    block-size: auto;
    inline-size: 100%;
  }

  .pythonrunner-container {
    border-radius: 10px;
    flex: 1;
    overflow: hidden;
    box-sizing: border-box;
  }

  .visual-output {
    box-sizing: border-box;
    overflow: auto;
  }
}

#app.--light {
  .embedded-viewer {
    background-color: $rpf-white;
  }
}
