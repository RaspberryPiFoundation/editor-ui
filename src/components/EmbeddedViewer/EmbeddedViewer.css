.embedded-container {
    padding-right: 2.5%;
    padding-left: 2.5%;
    max-height: 100vh;
}

.embedded-container .pythonrunner-console {
    min-height: None;
    padding-right: 0px;
}