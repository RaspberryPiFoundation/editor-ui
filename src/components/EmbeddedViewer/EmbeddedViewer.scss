@import '../../colours.scss';

.embedded-viewer {
  height: 100%;
  padding: var(--spacing-1);
  overflow: hidden;

  .pythonrunner-container {
    border-radius: 10px;
    overflow: hidden;
    box-sizing: border-box;

    .visual-output {
      box-sizing: border-box;
    }
  }
}

.--light {
  .embedded-viewer {
    background-color: $editor-light-turquoise;
    .pythonrunner-container {
      border: 2px solid $editor-light-grey;

      .visual-output {
        border-bottom: 5px solid $editor-light-grey;
      }
    }
  }
}

.--dark {
  .embedded-viewer {
    .pythonrunner-container {
      .visual-output {
        border-bottom: 5px solid $editor-grey;
      }
    }
  }
}
