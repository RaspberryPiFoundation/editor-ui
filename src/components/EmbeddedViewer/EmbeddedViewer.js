/* eslint-disable react-hooks/exhaustive-deps */
import './EmbeddedViewer.css';
import React from 'react';
import { useSelector } from 'react-redux'
import { useProject } from '../Editor/Hooks/useProject'
import { useEmbeddedMode } from '../Editor/Hooks/useEmbeddedMode'
import PythonRunner from '../Editor/Runners/PythonRunner/PythonRunner'
import EmbeddedControls from './EmbeddedControls/EmbeddedControls'

const EmbeddedViewer = (props) => {
  const projectLoaded = useSelector((state) => state.editor.projectLoaded);
  const projectIdentifier = props.match.params.identifier;
  useProject('python', projectIdentifier);
  useEmbeddedMode(true);

  return projectLoaded === true ? (
    <>
      <div className='embedded-container' >
        <EmbeddedControls />
        <div>
          <PythonRunner />
        </div>
      </div>
    </>
  ) : <p>Loading</p>;
};

export default EmbeddedViewer;
