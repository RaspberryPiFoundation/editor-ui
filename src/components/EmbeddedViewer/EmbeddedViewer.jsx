/* eslint-disable react-hooks/exhaustive-deps */
import "../../assets/stylesheets/EmbeddedViewer.scss";
import "../../assets/stylesheets/Project.scss";
import React, { useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import { useParams, useSearchParams } from "react-router-dom";

import { useProject } from "../../hooks/useProject";
import { useEmbeddedMode } from "../../hooks/useEmbeddedMode";
import { setBrowserPreview, setPage } from "../../redux/EditorSlice";
import Output from "../Editor/Output/Output";
import NotFoundModalEmbedded from "../Modals/NotFoundModalEmbedded";
import AccessDeniedNoAuthModalEmbedded from "../Modals/AccessDeniedNoAuthModalEmbedded";

const EmbeddedViewer = () => {
  const dispatch = useDispatch();

  const page = useSelector((state) => state.editor.page);
  const loading = useSelector((state) => state.editor.loading);
  const browserPreview = useSelector((state) => state.editor.browserPreview);
  const user = useSelector((state) => state.auth.user) || {};
  const notFoundModalShowing = useSelector(
    (state) => state.editor.notFoundModalShowing,
  );
  const accessDeniedNoAuthModalShowing = useSelector(
    (state) => state.editor.accessDeniedNoAuthModalShowing,
  );
  const { identifier } = useParams();
  const [searchParams, setSearchParams] = useSearchParams();

  useProject({
    projectIdentifier: identifier,
    accessToken: user.access_token,
    isEmbedded: true,
  });

  useEmbeddedMode(true);

  useEffect(() => {
    if (browserPreview) {
      setSearchParams({
        ...Object.fromEntries([...searchParams]),
        page,
      });
    }
  }, [page]);

  useEffect(() => {
    if (searchParams.get("browserPreview") === "true") {
      dispatch(setBrowserPreview(true));
    }

    if (searchParams.get("page")) {
      dispatch(setPage(searchParams.get("page")));
    }
  }, []);

  return (
    <div className="embedded-viewer">
      {loading === "success" ? <Output /> : null}
      {notFoundModalShowing ? <NotFoundModalEmbedded /> : null}
      {accessDeniedNoAuthModalShowing ? (
        <AccessDeniedNoAuthModalEmbedded />
      ) : null}
    </div>
  );
};

export default EmbeddedViewer;
