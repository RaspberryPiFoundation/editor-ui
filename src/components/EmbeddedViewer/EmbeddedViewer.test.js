import React from "react";
import EmbeddedViewer from "./EmbeddedViewer";

import { Provider } from "react-redux";
import configureStore from "redux-mock-store";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

let store;

beforeEach(() => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);

  const initialState = {
    editor: {
      loading: "success",
      project: {
        components: [],
      },
    },
    auth: {},
  };
  store = mockStore(initialState);
});

test("Renders without crashing", () => {
  const { asFragment } = render(
    <Provider store={store}>
      <MemoryRouter>
        <EmbeddedViewer />
      </MemoryRouter>
    </Provider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
