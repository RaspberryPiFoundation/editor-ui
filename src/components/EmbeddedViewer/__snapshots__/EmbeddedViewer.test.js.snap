// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<DocumentFragment>
  <div
    class="embedded-viewer"
  >
    <div
      hidden=""
      id="file-content"
    />
    <div
      class="proj-runner-container"
    >
      <div
        class="pythonrunner-container"
      >
        <div
          class="react-tabs"
          data-rttabs="true"
        >
          <ul
            class="react-tabs__tab-list"
            role="tablist"
          >
            <li
              aria-controls="react-tabs-1"
              aria-disabled="false"
              aria-selected="true"
              class="react-tabs__tab react-tabs__tab--selected"
              data-rttab="true"
              id="react-tabs-0"
              role="tab"
              tabindex="0"
            >
              Text Output
            </li>
          </ul>
          <div
            aria-labelledby="react-tabs-0"
            class="react-tabs__tab-panel react-tabs__tab-panel--selected"
            id="react-tabs-1"
            role="tabpanel"
          >
            <pre
              class="pythonrunner-console"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="runner-controls"
    >
      <button
        class="btn btn--run"
      >
        <svg
          fill="none"
          height="12"
          transform="scale(1, 1)"
          viewBox="0 0 12 12"
          width="12"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M1.25331 11.0714C1.12931 11.0714 1.01039 11.0262 0.922706 10.9458C0.835025 10.8655 0.785767 10.7564 0.785767 10.6428V1.21415C0.785781 1.13967 0.806965 1.06649 0.847232 1.0018C0.8875 0.937121 0.94546 0.883173 1.0154 0.845278C1.08534 0.807382 1.16485 0.786847 1.24608 0.785696C1.32732 0.784545 1.40748 0.802818 1.47866 0.838715L10.8294 5.55303C10.9027 5.59003 10.9639 5.64438 11.0064 5.71042C11.049 5.77646 11.0715 5.85176 11.0715 5.92847C11.0715 6.00517 11.049 6.08047 11.0064 6.14651C10.9639 6.21255 10.9027 6.2669 10.8294 6.3039L1.47866 11.0182C1.40963 11.0531 1.33211 11.0713 1.25331 11.0714Z"
            fill="#1A1A1A"
          />
        </svg>
        Run
      </button>
    </div>
  </div>
</DocumentFragment>
`;
